### PR TITLE
Enh: refactor semantic errors in ast_symboltable_visitor file

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: #5036

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
I've refactored the rest of the files too. Will push that after this gets merged.

**Comment by parth121101** [comment]
> I've refactored the rest of the files too (except common_visitor which @parth121101 is working on). Will push that after this gets merged.

@HarshitaKalani even common_visitor is refactored. Look PR 5403

**Comment by HarshitaKalani** [comment]
I was talking about the rest 15 you've mentioned here: https://github.com/lfortran/lfortran/pull/5403#issuecomment-2480688554

**Review by parth121101 — APPROVED** [review]

File: src/lfortran/semantics/ast_symboltable_visitor.cpp [hunk 1]
```diff
@@ -36,16 +36,25 @@ void extract_bind(T &x, ASR::abiType &abi_type, char *&bindc_name) {
                 } else if (to_lower(std::string(name->m_id)) == "js") {
                     abi_type=ASR::abiType::BindJS;
                 } else {
-                    throw SemanticError("Unsupported language in bind()",
-                        x.base.base.loc);
+                    diag.add(diag::Diagnostic(
+                        "Unsupported language in bind()",
+                        diag::Level::Error, diag::Stage::Semantic, {
+                            diag::Label("", {x.base.base.loc})}));
+                    throw SemanticAbort();
```
Comment (gxyd) [inline src/lfortran/semantics/ast_symboltable_visitor.cpp:43]: So, the idea is that with an incorrect specification of `bind` (e.g.; `bind(name=1)`) we raise only *one* error and not more than one?

As `bind(name=1)` has two possible errors for it, an unspecified binding specifier (like 'C') and incorrect value of `name` (as it only expects a string). So, we only intend to raise one error and exit out of that?

**Review by gxyd — COMMENTED** [review]
I'm not really always comfortable with change, unless I also see test cases for it. And I think it's possible with these changes that we might throw one error and exit out (as there is a use of `SemanticAbort()`) if my understand of the code here is correct.

Please correct me if I'm wrong in understanding here.

**Review by certik — APPROVED** [review]
It seems this PR is not changing any behavior. Only allowing (optionally) to continue compilation and getting rid of SemanticError, which we should get rid of completely. It seems it is behaving exactly as before.

**Comment by HarshitaKalani** [comment]
Yes @gxyd , the PR is to needed for error resiliency. Here, we're just getting rid of semanticError so that we can continue to compile even when there's error. I hope this clears things.

**Pull request merged**